RecoTracker/TkTrackingRegions/src/RectangularEtaPhiTrackingRegion.cc

```diff
@@ -47,6 +47,34 @@ namespace {
 using namespace PixelRecoUtilities;
 using namespace std;
 
+std::vector<bool> RectangularEtaPhiTrackingRegion::checkTracks(reco::TrackCollection const& InputCollection) const {
+  std::vector<bool> mask(InputCollection.size(), false);
+  int it = -1;
+  math::XYZPoint regOrigin(origin().x(), origin().y(), origin().z());
+  for (auto const& track : InputCollection) {
+    it++;
+    if (track.pt() < ptMin()) {
+      continue;
+    }
+
+    if (!etaRange().inside(track.eta())) {
+      continue;
+    }
+    if (std::abs(reco::deltaPhi(track.phi(), phiDirection())) > phiMargin().right()) {
```

Comment:
see above

Reply:
also, since `phiMargin()` could potentially be asymmetric, should this make explicit checks for direction - margin and direction + margin ?